datadog/data_source_datadog_rum_application.go

```diff
@@ -0,0 +1,109 @@
+package datadog
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceDatadogRUMApplication() *schema.Resource {
+	return &schema.Resource{
+		Description: "Use this data source to retrieve a Datadog RUM Application.",
+		ReadContext: dataSourceDatadogRUMApplicationRead,
+
+		Schema: map[string]*schema.Schema{
+			"name_filter": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name used to search for a RUM application",
+			},
+			"type_filter": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The type used to search for a RUM application",
+			},
+			"id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "ID of the RUM application. If set, this takes precedence over name and type filters.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The name of the RUM application",
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The RUM application type. Supported values are `browser`, `ios`, `android`, `react-native`, `flutter`",
+			},
+			"client_token": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The client token",
+			},
+		},
+	}
+}
+
+func dataSourceDatadogRUMApplicationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConf := meta.(*ProviderConfiguration)
+	apiInstances := providerConf.DatadogApiInstances
+	auth := providerConf.Auth
+
+	if searchID, ok := d.GetOk("id"); ok {
+		resp, _, err := apiInstances.GetRumApiV2().GetRUMApplication(auth, searchID.(string))
+		if err != nil {
+			return diag.Errorf("Couldn't find RUM application with id %s", searchID)
+		}
+
+		return dataSourceDatadogRUMApplicationUpdate(d, resp.Data.GetAttributes())
+	} else {
+		resp, _, err := apiInstances.GetRumApiV2().GetRUMApplications(auth)
+		if err != nil {
+			return diag.Errorf("Couldn't retrieve list of RUM Applications")
+		}
+
+		searchName, searchNameOk := d.GetOk("name_filter")
+		searchType, searchTypeOk := d.GetOk("type_filter")
+		var foundRUMApplicationIDs []string
+		for _, resp_data := range resp.Data {
+			if rum_app, ok := resp_data.GetAttributesOk(); ok {
+				nameSetAndMatched := searchNameOk && rum_app.GetName() == searchName
+				typeSetAndMatched := searchTypeOk && rum_app.GetType() == searchType
+				if nameSetAndMatched || typeSetAndMatched {
```

Comment:
I think we want the logic changed a bit here, because if type is set. we should match on both name and type, not just type.